### PR TITLE
Add Export table data logic to various tables

### PR DESCRIPTION
**Pull request by priyesh777**
Depends on : https://github.com/idmc-labs/helix-server/pull/193#issue-676164987

Remaining: 
- Add export feature for Actors
- Fix the query relative to "exportEntries" so that it would work for both the entries of extraction and nested entries.

## Changes
- Add file size info in downloads
- Add Export feature to Reports, Contacts, Organization and Figures

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/components/Downloads/DownloadedItem/index.tsx [hunk 1]
```diff
@@ -71,6 +73,9 @@ function DownloadedItem(props: DownloadedItemProps) {
                             transparent
                         />
                     )}
+                    <span>{fileSize && 'File Size:'}</span>
```
Comment (tnagorra) [inline src/components/Downloads/DownloadedItem/index.tsx:76]: I don't think we need a filesize label

File: src/components/tables/CommunicationAndPartners/index.tsx [hunk 2]
```diff
@@ -274,12 +342,23 @@ function CommunicationAndPartners(props: CommunicationAndPartnersProps) {
...
                     {contactPermissions?.add && (
-                        <Button
-                            name={undefined}
-                            onClick={showAddContactModal}
-                        >
-                            Add Contact
-                        </Button>
+                        <>
+                            <ConfirmButton
+                                confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/components/tables/CommunicationAndPartners/index.tsx:347]:
```suggestion
                                confirmationHeader={Export}
```

File: src/components/tables/EventsTable/index.tsx [hunk 3]
```diff
@@ -372,13 +373,15 @@ function EventsTable(props: EventsProps) {
             heading="Events"
             headerActions={(
                 <>
-                    <Button
+                    <ConfirmButton
+                        confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/components/tables/EventsTable/index.tsx:377]:
```suggestion
                        confirmationHeader="Confirm Export"
```

File: src/views/Countries/index.tsx [hunk 4]
```diff
@@ -222,13 +222,15 @@ function Countries(props: CountriesProps) {
                 className={styles.container}
                 contentClassName={styles.content}
                 headerActions={(
-                    <Button
+                    <ConfirmButton
+                        confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/views/Countries/index.tsx:226]:
```suggestion
                        confirmationHeader="Confirm Export"
```

File: src/views/Crises/index.tsx [hunk 5]
```diff
@@ -366,13 +367,15 @@ function Crises(props: CrisesProps) {
                 contentClassName={styles.content}
                 headerActions={(
                     <>
-                        <Button
+                        <ConfirmButton
+                            confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/views/Crises/index.tsx:371]:
```suggestion
                            confirmationHeader="Confirm Export"
```

File: src/views/Extraction/index.tsx [hunk 6]
```diff
@@ -330,13 +369,38 @@ function Extraction(props: ExtractionProps) {
...
+                                popupClassName={styles.popup}
+                                popupContentClassName={styles.popupContent}
+                                disabled={updateLoading || createLoading}
+                                label="Exports"
                             >
-                                Export
-                            </Button>
+                                <Row>
+                                    <ConfirmButton
+                                        confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/views/Extraction/index.tsx:383]:
```suggestion
                                        confirmationHeader="Confirm Export"
```

File: src/views/Extraction/index.tsx [hunk 7]
```diff
@@ -330,13 +369,38 @@ function Extraction(props: ExtractionProps) {
...
+                                        confirmationMessage="Are you sure you want to export Entries data ?"
+                                        name={undefined}
+                                        onConfirm={handleExportEntriesData}
+                                        disabled={exportingEntries}
+                                        variant="accent"
+                                    >
+                                        Export Entries
```
Comment (tnagorra) [inline src/views/Extraction/index.tsx:390]:
```suggestion
                                        Entries
```

File: src/views/Extraction/index.tsx [hunk 8]
```diff
@@ -330,13 +369,38 @@ function Extraction(props: ExtractionProps) {
...
+                                    </ConfirmButton>
+                                    <ConfirmButton
+                                        confirmationHeader={<h2>Confirm Export</h2>}
+                                        confirmationMessage="Are you sure you want to export Figures data ?"
+                                        name={undefined}
+                                        onConfirm={handleExportFiguresData}
+                                        disabled={exportingFigures}
+                                        variant="accent"
+                                    >
+                                        Export Figures
```
Comment (tnagorra) [inline src/views/Extraction/index.tsx:400]:
```suggestion
                                        Figures
```

File: src/views/Organizations/OrganizationTable/index.tsx [hunk 9]
```diff
@@ -256,12 +305,23 @@ function OrganizationTable(props: OrganizationProps) {
...
                     {orgPermissions?.add && (
-                        <Button
-                            name={undefined}
-                            onClick={showAddOrganizationModal}
-                        >
-                            Add Organization
-                        </Button>
+                        <>
+                            <ConfirmButton
+                                confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/views/Organizations/OrganizationTable/index.tsx:310]:
```suggestion
                                confirmationHeader="Confirm Export"
```

File: src/views/Reports/index.tsx [hunk 10]
```diff
@@ -306,13 +352,24 @@ function Reports(props: ReportsProps) {
...
-                            <Button
-                                name={undefined}
-                                onClick={showAddReportModal}
-                                disabled={loadingReports}
-                            >
-                                Add Report
-                            </Button>
+                            <>
+                                <ConfirmButton
+                                    confirmationHeader={<h2>Confirm Export</h2>}
```
Comment (tnagorra) [inline src/views/Reports/index.tsx:357]:
```suggestion
                                    confirmationHeader="Confirm Export"

File: src/components/Downloads/DownloadedItem/index.tsx [hunk 11]
```diff
@@ -71,6 +73,8 @@ function DownloadedItem(props: DownloadedItemProps) {
                             transparent
                         />
                     )}
+                    <span>{fileSize}</span>
+                    <span>{fileSize && 'KB'}</span>
```
Comment (tnagorra) [inline src/components/Downloads/DownloadedItem/index.tsx:77]: Are you sure this is in KB?

Comment (priyesh777) [reply src/components/Downloads/DownloadedItem/index.tsx:77]: That was an assumption..but now I think this is in bytes

Comment (tnagorra) [reply src/components/Downloads/DownloadedItem/index.tsx]: It's mostly in bytes :v:

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**